crates/trigger-http/src/lib.rs

```diff
@@ -490,6 +490,9 @@ impl OutboundWasiHttpHandler for HttpRuntimeData {
                 .parse()
                 // origin together with the path and query must be a valid URI
                 .unwrap();
+            let host = format!("{}:{}", uri.host().unwrap(), uri.port().unwrap());
+            let headers = request.request.headers_mut();
+            headers.insert(HOST, HeaderValue::from_str(&host)?);
```

Comment:
Same question about potentially overwriting an existing host header.

Reply:
We are only overwriting on self-requests.

Reply:
I understand that but what if some mad, bad person sends a self request and _explicitly sets the host header on it._  (I cannot imagine why they would ever do this... I just wonder if we should respect it if they do.)

Reply:
I will check if it is a case of overwriting always or only if the user explicitly sets it. In that case, we can check for its existence before setting it.

Reply:
By default, the host seems to be 
```
host = Some(
    ":443",
)
```

Do we want to do something like 
```
if let Some(":443") = headers.get("host") {
    headers.insert(HOST, HeaderValue::from_str(&host)?);
}
```

Reply:
Egad!  That host looks extremely weird, though I guess it's what we should expect given what we saw in the issue.  It makes me think it's already getting set _somewhere_ - and that makes me wonder if we can tweak wherever it's currently getting set rather than overwriting it later in the pipeline.  It reminded me of the issue @dicej found (https://github.com/fermyon/spin-rust-sdk/issues/8#issuecomment-1961575084) so there may be some subtle difference between main and RTM anyway...

Anyway, "user explicitly sets Host header in a self-request" certainly feels like a wildly obscure edge case.  Given that it's not trivial to determine, I'm happy to punt on it for the time being.  Thanks for the detailed investigation!

Reply:
I'll try to dig into where the header actually gets set. I will also update the rust SDK that the app uses to  `main` instead of pinning it to `v2.2` to see if that interaction differs.

Reply:
I did some digging and this originates from the parsing of the `uri` by `hyper`.  The first request goes through fine because the `host` is set properly on the incoming request that we use that to construct the URL and then in the subsequent requests, it fails. 

ref: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=fb83cc1730a01a63f2d9c70ff95074a8